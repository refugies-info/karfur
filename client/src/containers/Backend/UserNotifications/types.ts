import { Id } from "api-types";

export interface FormattedNotification {
  type: "reaction" | "annuaire" | "new content";
  read: boolean;
  link?: string;
  username?: string;
  createdAt?: Date;
  suggestionId?: string;
  text?: string;
  title?: string;
  dispositifId?: Id;
  typeContenu?: "dispositif" | "demarche";
}
