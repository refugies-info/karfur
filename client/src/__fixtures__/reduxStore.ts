import { RootState } from "services/rootReducer";
import { SearchResultsState } from "services/SearchResults/searchResults.reducer";
import { UserFavoritesState } from "services/UserFavoritesInLocale/UserFavoritesInLocale.reducer";
import { activeThemesMock } from "./activeThemes";

const initialMockUserState = {
  user: null,
  admin: false,
  traducteur: false,
  expertTrad: false,
  contributeur: false,
  membreStruct: false,
  userId: "",
  userFetched: false,
  rolesInStructure: [],
};
const initialMockLangueState = {
  langues: [],
  languei18nCode: "fr",
  showLanguageModal: false,
  showLangModal: false,
};

const initialMockTranslationState = {
  translation: {
    initialText: {},
    translatedText: {},
  },
  translations: [],
};

const initialUserFavoritesState: UserFavoritesState = {
  favorites: [],
  showFavoriteModal: false
}

const initialMockSearchReults: SearchResultsState = {
  results: {
    dispositifs: [],
    demarches: [],
    dispositifsSecondaryTheme: []
  },
  query: {
    search: "",
    departments: [],
    themes: [],
    needs: [],
    age: [],
    frenchLevel: [],
    language: [],
    sort: "date",
    type: "all",
  }
}

export const initialMockStore = {
  user: initialMockUserState,
  langue: initialMockLangueState,
  activeDispositifs: [],
  tts: { ttsActive: false, showAudioSpinner: false },
  userStructure: null,
  selectedDispositif: null,
  loadingStatus: {},
  translation: initialMockTranslationState,
  activeStructures: [],
  selectedStructure: null,
  allDispositifs: [],
  allStructures: [],
  users: [],
  userFavorites: initialUserFavoritesState,
  userContributions: [],
  needs: [],
  themes: {
    activeThemes: activeThemesMock,
    inactiveThemes: []
  },
  searchResults: initialMockSearchReults
};
