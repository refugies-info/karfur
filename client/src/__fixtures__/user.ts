import { GetUserInfoResponse, UserStatus } from "api-types";
import moment from "moment";

const date = moment("2019-05-01 12:00:00");
export const testUser: GetUserInfoResponse = {
  username: "test",
  _id: "testObjectId",
  structures: [],
  contributions: [],
  phone: "",
  roles: [],
  email: "",
  selectedLanguages: [],
  status: UserStatus.ACTIVE,
  traductionsFaites: []
};

export const testUserWithRoles: GetUserInfoResponse = {
  ...testUser,
  structures: ["testObjectId"],
  roles: [
    { nom: "Admin", _id: "testObjectId", nomPublic: "Admin" },
    { nom: "Trad", _id: "testObjectId", nomPublic: "Trad" },
    {
      nom: "ExpertTrad",
      _id: "testObjectId",
      nomPublic: "ExpertTrad"
    },
    {
      nom: "Contrib",
      _id: "testObjectId",
      nomPublic: "Contrib"
    }
  ]
};
