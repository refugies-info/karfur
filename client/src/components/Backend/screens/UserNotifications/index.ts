import UserNotifications from "./UserNotifications";

export default UserNotifications;
