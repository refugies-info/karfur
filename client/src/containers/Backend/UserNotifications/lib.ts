import { GetDispositifsResponse } from "api-types";
import { FormattedNotification } from "./types";

export const formatNotifications = (
  dispositifs: GetDispositifsResponse[],
  hasResponsibleSeenAnnuaireNotif: boolean
): FormattedNotification[] => {
  let result: FormattedNotification[] = [];
  if (!hasResponsibleSeenAnnuaireNotif) {
    result.push({ type: "annuaire", read: false });
  }
  if (!dispositifs) return [];
  dispositifs.forEach((dispo) => {
    if (dispo.status === "En attente") {
      result.push({
        type: "new content",
        link: "/" + dispo.typeContenu + "/" + dispo._id,
        read: false,
        createdAt: dispo.created_at,
      });
    }
    /* TODO: add suggestions to dispositif?
    if (!dispo.suggestions) return;
    dispo.suggestions.forEach((suggestion) => {
      result.push({
        type: "reaction",
        username: suggestion.username,
        createdAt: suggestion.createdAt,
        suggestionId: suggestion.suggestionId,
        read: !!suggestion.read,
        text: suggestion.suggestion,
        title: dispo.titreInformatif,
        dispositifId: dispo._id,
        typeContenu: dispo.typeContenu,
      });
    }); */
  });
  result.sort((a, b) => {
    if (a.read && !b.read) {
      return 1;
    }
    if (!a.read && b.read) {
      return -1;
    }
    if (!a.read && !b.read) {
      if (a.type === "annuaire") {
        return -1;
      }
      if (b.type === "annuaire") {
        return 1;
      }
      if (
        a.type === "new content" &&
        b.type === "new content" &&
        a.createdAt &&
        b.createdAt &&
        a.createdAt < b.createdAt
      ) {
        return 1;
      }

      if (a.type === "new content") {
        return -1;
      }
      if (b.type === "new content") {
        return 1;
      }

      if (a.createdAt && b.createdAt && a.createdAt > b.createdAt) return -1;
      return 1;
    }
    //trie par date les notifs lues
    if (a.createdAt && b.createdAt && a.createdAt > b.createdAt) return -1;
    return 1;
  });
  return result;
};

export const getNbNewNotifications = (
  dispositifs: GetDispositifsResponse[],
  hasResponsibleSeenAnnuaireNotif: boolean
) => {
  const formattedNotifications = formatNotifications(
    dispositifs,
    hasResponsibleSeenAnnuaireNotif
  );
  return formattedNotifications.filter((notif) => !notif.read).length;
};
