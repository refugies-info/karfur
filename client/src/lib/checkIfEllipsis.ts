export const checkIfEllipsis = (element: HTMLElement) => element.offsetWidth < element.scrollWidth;
