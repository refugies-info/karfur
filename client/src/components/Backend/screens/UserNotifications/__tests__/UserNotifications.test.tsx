// @ts-nocheck
import "jest-styled-components";
import Router from "next/router";
import { updateDispositifReactionActionCreator } from "services/ActiveDispositifs/activeDispositifs.actions";
import { fetchSelectedStructureActionCreator } from "services/SelectedStructure/selectedStructure.actions";
import { fetchUserStructureActionCreator } from "services/UserStructure/userStructure.actions";
import { initialMockStore } from "__fixtures__/reduxStore";
import { wrapWithProvidersAndRenderForTesting } from "../../../../../../jest/lib/wrapWithProvidersAndRender";
import UserNotifications from "../UserNotifications";
jest.mock("next/router", () => require("next-router-mock"));

jest.mock("services/UserStructure/userStructure.actions", () => {
  const actions = jest.requireActual("services/UserStructure/userStructure.actions");
  return {
    fetchUserStructureActionCreator: jest.fn(actions.fetchUserStructureActionCreator),
    updateUserStructureActionCreator: jest.fn(actions.updateUserStructureActionCreator),
  };
});

jest.mock("services/ActiveDispositifs/activeDispositifs.actions", () => {
  const actions = jest.requireActual("services/ActiveDispositifs/activeDispositifs.actions");
  return {
    updateDispositifReactionActionCreator: jest.fn(actions.updateDispositifReactionActionCreator),
  };
});

jest.mock("services/SelectedStructure/selectedStructure.actions", () => {
  const actions = jest.requireActual("services/SelectedStructure/selectedStructure.actions");
  return {
    fetchSelectedStructureActionCreator: jest.fn(actions.fetchSelectedStructureActionCreator),
  };
});

describe("UserNotifications", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  it("should render correctly when loading", () => {
    window.scrollTo = jest.fn();
    const { asFragment } = wrapWithProvidersAndRenderForTesting({
      Component: UserNotifications,
      reduxState: {
        ...initialMockStore,
        loadingStatus: { FETCH_USER_STRUCTURE: { isLoading: true } },
        user: { user: { structures: ["structureId"] } },
      },
    });

    expect(fetchUserStructureActionCreator).toHaveBeenCalledWith({
      structureId: "structureId",
      shouldRedirect: false,
    });
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render correctly when 0 notif", () => {
    window.scrollTo = jest.fn();
    const { asFragment } = wrapWithProvidersAndRenderForTesting({
      Component: UserNotifications,
      reduxState: {
        ...initialMockStore,
        loadingStatus: { FETCH_USER_STRUCTURE: { isLoading: false } },
        user: { user: { structures: ["structureId"] } },
        userStructure: {
          _id: 1,
          hasResponsibleSeenNotification: true,
          dispositifsAssocies: [],
        },
      },
    });
    expect(fetchUserStructureActionCreator).toHaveBeenCalledWith({
      structureId: "structureId",
      shouldRedirect: false,
    });
    expect(fetchSelectedStructureActionCreator).toHaveBeenCalledWith({
      id: "1",
      locale: "fr",
    });
    expect(asFragment()).toMatchSnapshot();
  });

  const notifNewContent = {
    status: "En attente",
    typeContenu: "dispositif",
    _id: "id",
    created_at: "01/01/2021",
  };

  const notifReactionNotRead = {
    suggestions: [
      {
        username: "user1",
        createdAt: "02/01/2021",
        suggestionId: "suggesID",
        suggestion: "test reaction non lue",
      },
    ],
    titreInformatif: "titre info",
    _id: "id1",
    status: "Actif",
  };

  const notifReactionRead = {
    suggestions: [
      {
        username: "user2",
        createdAt: "02/02/2022",
        suggestionId: "suggesID2",
        suggestion: "test reaction lue",
        read: true,
      },
    ],
    titreInformatif: "titre info2",
    _id: "id2",
    status: "Actif",
  };

  it("should render correctly when notif not read annuaire, not read reaction, not read new content, read reaction ", () => {
    window.scrollTo = jest.fn();
    const { asFragment } = wrapWithProvidersAndRenderForTesting({
      Component: UserNotifications,
      reduxState: {
        ...initialMockStore,
        loadingStatus: { FETCH_USER_STRUCTURE: { isLoading: false } },
        user: { user: { structures: ["structureId"] } },
        userStructure: {
          _id: 1,
          hasResponsibleSeenNotification: false,
          dispositifsAssocies: [notifNewContent, notifReactionRead, notifReactionNotRead],
        },
      },
    });
    expect(fetchUserStructureActionCreator).toHaveBeenCalledWith({
      structureId: "structureId",
      shouldRedirect: false,
    });
    expect(fetchSelectedStructureActionCreator).toHaveBeenCalledWith({
      id: "1",
      locale: "fr",
    });
    expect(asFragment()).toMatchSnapshot();
  });

  it("should link to dispositif when click on new content notif", () => {
    window.scrollTo = jest.fn();
    const component = wrapWithProvidersAndRenderForTesting({
      Component: UserNotifications,
      reduxState: {
        ...initialMockStore,
        loadingStatus: { FETCH_USER_STRUCTURE: { isLoading: false } },
        user: { user: { structures: ["structureId"] } },
        userStructure: {
          _id: 1,
          hasResponsibleSeenNotification: false,
          dispositifsAssocies: [notifNewContent],
        },
      },
    });
    expect(fetchUserStructureActionCreator).toHaveBeenCalledWith({
      structureId: "structureId",
      shouldRedirect: false,
    });
    expect(fetchSelectedStructureActionCreator).toHaveBeenCalledWith({
      id: "1",
      locale: "fr",
    });
    component.root
      .findByProps({ "data-testid": "test-notif-new content" })
      .props.onClick({ stopPropagation: jest.fn() });
    expect(Router).toMatchObject({ asPath: "/dispositif/id" });
  });

  it("should link to dispositif when click on annuaire notif", () => {
    window.scrollTo = jest.fn();
    const component = wrapWithProvidersAndRenderForTesting({
      Component: UserNotifications,
      reduxState: {
        ...initialMockStore,
        loadingStatus: { FETCH_USER_STRUCTURE: { isLoading: false } },
        user: { user: { structures: ["structureId"] } },
        userStructure: {
          _id: 1,
          hasResponsibleSeenNotification: false,
          dispositifsAssocies: [],
        },
      },
    });
    expect(fetchUserStructureActionCreator).toHaveBeenCalledWith({
      structureId: "structureId",
      shouldRedirect: false,
    });
    expect(fetchSelectedStructureActionCreator).toHaveBeenCalledWith({
      id: "1",
      locale: "fr",
    });
    component.root.findByProps({ "data-testid": "test-notif-annuaire" }).props.onClick({ stopPropagation: jest.fn() });
    expect(Router).toMatchObject({ asPath: "/directory-create" });
  });

  it("should delete notif reaction", () => {
    window.scrollTo = jest.fn();
    const component = wrapWithProvidersAndRenderForTesting({
      Component: UserNotifications,
      reduxState: {
        ...initialMockStore,
        loadingStatus: { FETCH_USER_STRUCTURE: { isLoading: false } },
        user: { user: { structures: ["structureId"] } },
        userStructure: {
          _id: 1,
          hasResponsibleSeenNotification: true,
          dispositifsAssocies: [notifReactionRead],
        },
      },
    });
    expect(fetchUserStructureActionCreator).toHaveBeenCalledWith({
      structureId: "structureId",
      shouldRedirect: false,
    });
    expect(fetchSelectedStructureActionCreator).toHaveBeenCalledWith({
      id: "1",
      locale: "fr",
    });
    component.root.findByProps({ "data-testid": "test-delete-reaction" }).props.onClick({ stopPropagation: jest.fn() });
    expect(updateDispositifReactionActionCreator).toHaveBeenLastCalledWith({
      suggestion: {
        dispositifId: "id2",
        suggestionId: "suggesID2",
        type: "remove",
      },
      structureId: "structureId",
    });
  });
});
