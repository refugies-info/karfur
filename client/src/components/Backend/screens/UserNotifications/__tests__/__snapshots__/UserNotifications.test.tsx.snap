// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserNotifications should render correctly when 0 notif 1`] = `
.c0 {
  background: #edebeb;
  border-radius: 12px;
  padding: 40px;
  margin-top: 26px;
  margin-right: 120px;
  margin-left: 120px;
  height: fit-content;
}

.c2 {
  font-weight: bold;
  font-size: 28px;
  line-height: 35px;
  margin-bottom: 8px;
}

.c1 {
  color: #5e5e5e;
  margin-top: 20px;
  display: flex;
  flex-direction: column;
  align-items: center;
}

<div
  style={
    {
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <div
    className="c0"
  >
    <div
      className="c1"
    >
      <div
        className="c2"
      >
        Aucune notification ! 
      </div>
      <div
        style={
          {
            "marginBottom": "20px",
            "marginTop": "32px",
          }
        }
      >
        <img
          alt="no-notification"
          data-nimg="1"
          decoding="async"
          height={289}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="https://storage.googleapis.com/refugies-info-assets/middle-office/aucune_notification.svg"
          style={
            {
              "color": "transparent",
            }
          }
          width={400}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`UserNotifications should render correctly when loading 1`] = `
.c1 {
  display: flex;
  flex-direction: row;
  margin-bottom: 32px;
  align-items: flex-end;
}

.c3 {
  background: #212121;
  border-radius: 12px;
  font-weight: bold;
  font-size: 28px;
  line-height: 35px;
  color: #ffffff;
  padding: 8px 19px 8px 19px;
  margin: 0px 8px 0px 8px;
}

.c2 {
  font-weight: bold;
  font-size: 28px;
  line-height: 35px;
  margin-bottom: 8px;
}

.c0 {
  background: #edebeb;
  border-radius: 12px;
  padding: 40px;
  margin-top: 26px;
  margin-right: 120px;
  margin-left: 120px;
  height: fit-content;
}

<div
  style={
    {
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <div
    className="c0"
  >
    <div
      className="c1"
    >
      <div
        className="c2"
      >
        Vous avez
      </div>
      <div
        className="c3"
      >
        ...
      </div>
      <div
        className="c2"
      >
        nouvelle notification.
      </div>
    </div>
    <span
      aria-busy={true}
      aria-live="polite"
    >
      <span
        className="react-loading-skeleton"
        style={
          {
            "height": 50,
          }
        }
      >
        ‌
      </span>
      <br />
      <span
        className="react-loading-skeleton"
        style={
          {
            "height": 50,
          }
        }
      >
        ‌
      </span>
      <br />
      <span
        className="react-loading-skeleton"
        style={
          {
            "height": 50,
          }
        }
      >
        ‌
      </span>
      <br />
    </span>
  </div>
</div>
`;

exports[`UserNotifications should render correctly when notif not read annuaire, not read reaction, not read new content, read reaction  1`] = `
.c4 {
  background: #2d9cdb;
  border-radius: 12px;
  padding: 8px 8px 8px 20px;
  margin: 8px 0px 0px 0px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  cursor: pointer;
  border-width: 2px;
  border-style: solid;
  border-color: #2d9cdb;
}

.c4:hover {
  border-color: #212121;
}

.c9 {
  background: #ffffff;
  border-radius: 12px;
  padding: 8px 8px 8px 20px;
  margin: 8px 0px 0px 0px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  cursor: pointer;
  border-width: 2px;
  border-style: solid;
  border-color: #ffffff;
}

.c9:hover {
  border-color: #212121;
}

.c5 {
  display: flex;
  flex-direction: row;
  align-items: center;
}

.c6 {
  font-weight: bold;
  font-size: 18px;
  line-height: 23px;
  color: #ffffff;
  margin-left: 20px;
}

.c10 {
  font-weight: bold;
  font-size: 18px;
  line-height: 23px;
  color: #212121;
  margin-left: 20px;
}

.c8 {
  background: #edebeb;
  border-radius: 8px;
  padding: 8px;
  font-weight: bold;
  font-size: 18px;
  line-height: 23px;
  margin-left: 8px;
}

.c7 {
  font-weight: bold;
  font-size: 16px;
  line-height: 20px;
  color: #ffffff;
  margin-right: 8px;
  margin-left: 8px;
}

.c1 {
  display: flex;
  flex-direction: row;
  margin-bottom: 32px;
  align-items: flex-end;
}

.c3 {
  background: #212121;
  border-radius: 12px;
  font-weight: bold;
  font-size: 28px;
  line-height: 35px;
  color: #ffffff;
  padding: 8px 19px 8px 19px;
  margin: 0px 8px 0px 8px;
}

.c2 {
  font-weight: bold;
  font-size: 28px;
  line-height: 35px;
  margin-bottom: 8px;
}

.c0 {
  background: #edebeb;
  border-radius: 12px;
  padding: 40px;
  margin-top: 26px;
  margin-right: 120px;
  margin-left: 120px;
  height: fit-content;
}

<div
  style={
    {
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <div
    className="c0"
  >
    <div
      className="c1"
    >
      <div
        className="c2"
      >
        Vous avez
      </div>
      <div
        className="c3"
      >
        3
      </div>
      <div
        className="c2"
      >
        nouvelles notifications.
      </div>
    </div>
    <div
      className="c4"
      data-test-id="test-notif-annuaire"
      onClick={[Function]}
      read={false}
    >
      <div
        className="c5"
      >
        <span
          className="icon"
        >
          <i
            dangerouslySetInnerHTML={
              {
                "__html": "<svg xmlns="http://www.w3.org/2000/svg" width="20px" height="20px" viewBox="0 0 24 24" class="eva eva-bell" fill="#ffffff"><g data-name="Layer 2"><g data-name="bell"><rect width="24" height="24" opacity="0"/><path d="M20.52 15.21l-1.8-1.81V8.94a6.86 6.86 0 0 0-5.82-6.88 6.74 6.74 0 0 0-7.62 6.67v4.67l-1.8 1.81A1.64 1.64 0 0 0 4.64 18H8v.34A3.84 3.84 0 0 0 12 22a3.84 3.84 0 0 0 4-3.66V18h3.36a1.64 1.64 0 0 0 1.16-2.79zM14 18.34A1.88 1.88 0 0 1 12 20a1.88 1.88 0 0 1-2-1.66V18h4z"/></g></g></svg>",
              }
            }
            style={
              {
                "height": "20px",
                "lineHeight": "20px",
              }
            }
          />
        </span>
        <div
          className="c6"
          read={false}
        >
          Recensez votre structure dans l'annuaire de l'intégration
        </div>
      </div>
      <div
        className="c5"
      >
        <button
          className="btn dark me-2"
        >
          <span
            className="me-2 icon"
          >
            <i
              dangerouslySetInnerHTML={
                {
                  "__html": "<svg xmlns="http://www.w3.org/2000/svg" width="20px" height="20px" viewBox="0 0 24 24" class="eva eva-folder-add-outline" fill="#fff"><g data-name="Layer 2"><g data-name="folder-add"><rect width="24" height="24" opacity="0"/><path d="M14 13h-1v-1a1 1 0 0 0-2 0v1h-1a1 1 0 0 0 0 2h1v1a1 1 0 0 0 2 0v-1h1a1 1 0 0 0 0-2z"/><path d="M19.5 7.05h-7L9.87 3.87a1 1 0 0 0-.77-.37H4.5A2.47 2.47 0 0 0 2 5.93v12.14a2.47 2.47 0 0 0 2.5 2.43h15a2.47 2.47 0 0 0 2.5-2.43V9.48a2.47 2.47 0 0 0-2.5-2.43zm.5 11a.46.46 0 0 1-.5.43h-15a.46.46 0 0 1-.5-.43V5.93a.46.46 0 0 1 .5-.43h4.13l2.6 3.18a1 1 0 0 0 .77.37h7.5a.46.46 0 0 1 .5.43z"/></g></g></svg>",
                }
              }
              style={
                {
                  "height": "20px",
                  "lineHeight": "20px",
                }
              }
            />
          </span>
          Compléter la fiche annuaire
        </button>
        <button
          className="btn error ms-2"
          data-test-id="test-delete-annuaire"
          onClick={[Function]}
        >
          <span
            className="icon"
          >
            <i
              dangerouslySetInnerHTML={
                {
                  "__html": "<svg xmlns="http://www.w3.org/2000/svg" width="20px" height="20px" viewBox="0 0 24 24" class="eva eva-trash-2" fill="#fff"><g data-name="Layer 2"><g data-name="trash-2"><rect width="24" height="24" opacity="0"/><path d="M21 6h-5V4.33A2.42 2.42 0 0 0 13.5 2h-3A2.42 2.42 0 0 0 8 4.33V6H3a1 1 0 0 0 0 2h1v11a3 3 0 0 0 3 3h10a3 3 0 0 0 3-3V8h1a1 1 0 0 0 0-2zM10 16a1 1 0 0 1-2 0v-4a1 1 0 0 1 2 0zm0-11.67c0-.16.21-.33.5-.33h3c.29 0 .5.17.5.33V6h-4zM16 16a1 1 0 0 1-2 0v-4a1 1 0 0 1 2 0z"/></g></g></svg>",
                }
              }
              style={
                {
                  "height": "20px",
                  "lineHeight": "20px",
                }
              }
            />
          </span>
        </button>
      </div>
    </div>
    <div
      className="c4"
      data-test-id="test-notif-new content"
      onClick={[Function]}
      read={false}
    >
      <div
        className="c5"
      >
        <span
          className="icon"
        >
          <i
            dangerouslySetInnerHTML={
              {
                "__html": "<svg xmlns="http://www.w3.org/2000/svg" width="20px" height="20px" viewBox="0 0 24 24" class="eva eva-bell" fill="#ffffff"><g data-name="Layer 2"><g data-name="bell"><rect width="24" height="24" opacity="0"/><path d="M20.52 15.21l-1.8-1.81V8.94a6.86 6.86 0 0 0-5.82-6.88 6.74 6.74 0 0 0-7.62 6.67v4.67l-1.8 1.81A1.64 1.64 0 0 0 4.64 18H8v.34A3.84 3.84 0 0 0 12 22a3.84 3.84 0 0 0 4-3.66V18h3.36a1.64 1.64 0 0 0 1.16-2.79zM14 18.34A1.88 1.88 0 0 1 12 20a1.88 1.88 0 0 1-2-1.66V18h4z"/></g></g></svg>",
              }
            }
            style={
              {
                "height": "20px",
                "lineHeight": "20px",
              }
            }
          />
        </span>
        <div
          className="c6"
          read={false}
        >
          Une nouvelle fiche a été attribuée à votre structure
        </div>
      </div>
      <div
        className="c5"
      >
        <div
          className="c7"
          read={false}
        >
          Depuis 5 jours
        </div>
        <button
          className="btn dark"
        >
          <span
            className="me-2 icon"
          >
            <i
              dangerouslySetInnerHTML={
                {
                  "__html": "<svg xmlns="http://www.w3.org/2000/svg" width="20px" height="20px" viewBox="0 0 24 24" class="eva eva-eye" fill="#fff"><g data-name="Layer 2"><g data-name="eye"><rect width="24" height="24" opacity="0"/><circle cx="12" cy="12" r="1.5"/><path d="M21.87 11.5c-.64-1.11-4.16-6.68-10.14-6.5-5.53.14-8.73 5-9.6 6.5a1 1 0 0 0 0 1c.63 1.09 4 6.5 9.89 6.5h.25c5.53-.14 8.74-5 9.6-6.5a1 1 0 0 0 0-1zm-9.87 4a3.5 3.5 0 1 1 3.5-3.5 3.5 3.5 0 0 1-3.5 3.5z"/></g></g></svg>",
                }
              }
              style={
                {
                  "height": "20px",
                  "lineHeight": "20px",
                }
              }
            />
          </span>
          Voir la fiche
        </button>
      </div>
    </div>
    <div
      className="c4"
      data-test-id="test-notif-reaction"
      onClick={[Function]}
    >
      <div
        className="c5"
      >
        <span
          className="icon"
        >
          <i
            dangerouslySetInnerHTML={
              {
                "__html": "<svg xmlns="http://www.w3.org/2000/svg" width="20px" height="20px" viewBox="0 0 24 24" class="eva eva-bell" fill="#ffffff"><g data-name="Layer 2"><g data-name="bell"><rect width="24" height="24" opacity="0"/><path d="M20.52 15.21l-1.8-1.81V8.94a6.86 6.86 0 0 0-5.82-6.88 6.74 6.74 0 0 0-7.62 6.67v4.67l-1.8 1.81A1.64 1.64 0 0 0 4.64 18H8v.34A3.84 3.84 0 0 0 12 22a3.84 3.84 0 0 0 4-3.66V18h3.36a1.64 1.64 0 0 0 1.16-2.79zM14 18.34A1.88 1.88 0 0 1 12 20a1.88 1.88 0 0 1-2-1.66V18h4z"/></g></g></svg>",
              }
            }
            style={
              {
                "height": "20px",
                "lineHeight": "20px",
              }
            }
          />
        </span>
        <div
          className="c6"
        >
          Nouvelle réaction sur la fiche :
        </div>
        <div
          className="c8"
        >
          titre info
        </div>
      </div>
      <div
        className="c5"
      >
        <button
          className="btn dark"
        >
          <span
            className="me-2 icon"
          >
            <i
              dangerouslySetInnerHTML={
                {
                  "__html": "<svg xmlns="http://www.w3.org/2000/svg" width="20px" height="20px" viewBox="0 0 24 24" class="eva eva-eye" fill="#fff"><g data-name="Layer 2"><g data-name="eye"><rect width="24" height="24" opacity="0"/><circle cx="12" cy="12" r="1.5"/><path d="M21.87 11.5c-.64-1.11-4.16-6.68-10.14-6.5-5.53.14-8.73 5-9.6 6.5a1 1 0 0 0 0 1c.63 1.09 4 6.5 9.89 6.5h.25c5.53-.14 8.74-5 9.6-6.5a1 1 0 0 0 0-1zm-9.87 4a3.5 3.5 0 1 1 3.5-3.5 3.5 3.5 0 0 1-3.5 3.5z"/></g></g></svg>",
                }
              }
              style={
                {
                  "height": "20px",
                  "lineHeight": "20px",
                }
              }
            />
          </span>
          Voir la réaction
        </button>
        <button
          className="btn error ms-2"
          data-test-id="test-delete-reaction"
          onClick={[Function]}
        >
          <span
            className="icon"
          >
            <i
              dangerouslySetInnerHTML={
                {
                  "__html": "<svg xmlns="http://www.w3.org/2000/svg" width="20px" height="20px" viewBox="0 0 24 24" class="eva eva-trash-2" fill="#fff"><g data-name="Layer 2"><g data-name="trash-2"><rect width="24" height="24" opacity="0"/><path d="M21 6h-5V4.33A2.42 2.42 0 0 0 13.5 2h-3A2.42 2.42 0 0 0 8 4.33V6H3a1 1 0 0 0 0 2h1v11a3 3 0 0 0 3 3h10a3 3 0 0 0 3-3V8h1a1 1 0 0 0 0-2zM10 16a1 1 0 0 1-2 0v-4a1 1 0 0 1 2 0zm0-11.67c0-.16.21-.33.5-.33h3c.29 0 .5.17.5.33V6h-4zM16 16a1 1 0 0 1-2 0v-4a1 1 0 0 1 2 0z"/></g></g></svg>",
                }
              }
              style={
                {
                  "height": "20px",
                  "lineHeight": "20px",
                }
              }
            />
          </span>
        </button>
      </div>
    </div>
    <div
      className="c9"
      data-test-id="test-notif-reaction"
      onClick={[Function]}
      read={true}
    >
      <div
        className="c5"
      >
        <span
          className="icon"
        >
          <i
            dangerouslySetInnerHTML={
              {
                "__html": "<svg xmlns="http://www.w3.org/2000/svg" width="20px" height="20px" viewBox="0 0 24 24" class="eva eva-bell-outline" fill="#212121"><g data-name="Layer 2"><g data-name="bell"><rect width="24" height="24" opacity="0"/><path d="M20.52 15.21l-1.8-1.81V8.94a6.86 6.86 0 0 0-5.82-6.88 6.74 6.74 0 0 0-7.62 6.67v4.67l-1.8 1.81A1.64 1.64 0 0 0 4.64 18H8v.34A3.84 3.84 0 0 0 12 22a3.84 3.84 0 0 0 4-3.66V18h3.36a1.64 1.64 0 0 0 1.16-2.79zM14 18.34A1.88 1.88 0 0 1 12 20a1.88 1.88 0 0 1-2-1.66V18h4zM5.51 16l1.18-1.18a2 2 0 0 0 .59-1.42V8.73A4.73 4.73 0 0 1 8.9 5.17 4.67 4.67 0 0 1 12.64 4a4.86 4.86 0 0 1 4.08 4.9v4.5a2 2 0 0 0 .58 1.42L18.49 16z"/></g></g></svg>",
              }
            }
            style={
              {
                "height": "20px",
                "lineHeight": "20px",
              }
            }
          />
        </span>
        <div
          className="c10"
          read={true}
        >
          Nouvelle réaction sur la fiche :
        </div>
        <div
          className="c8"
        >
          titre info2
        </div>
      </div>
      <div
        className="c5"
      >
        <button
          className="btn dark"
        >
          <span
            className="me-2 icon"
          >
            <i
              dangerouslySetInnerHTML={
                {
                  "__html": "<svg xmlns="http://www.w3.org/2000/svg" width="20px" height="20px" viewBox="0 0 24 24" class="eva eva-eye" fill="#fff"><g data-name="Layer 2"><g data-name="eye"><rect width="24" height="24" opacity="0"/><circle cx="12" cy="12" r="1.5"/><path d="M21.87 11.5c-.64-1.11-4.16-6.68-10.14-6.5-5.53.14-8.73 5-9.6 6.5a1 1 0 0 0 0 1c.63 1.09 4 6.5 9.89 6.5h.25c5.53-.14 8.74-5 9.6-6.5a1 1 0 0 0 0-1zm-9.87 4a3.5 3.5 0 1 1 3.5-3.5 3.5 3.5 0 0 1-3.5 3.5z"/></g></g></svg>",
                }
              }
              style={
                {
                  "height": "20px",
                  "lineHeight": "20px",
                }
              }
            />
          </span>
          Voir la réaction
        </button>
        <button
          className="btn error ms-2"
          data-test-id="test-delete-reaction"
          onClick={[Function]}
        >
          <span
            className="icon"
          >
            <i
              dangerouslySetInnerHTML={
                {
                  "__html": "<svg xmlns="http://www.w3.org/2000/svg" width="20px" height="20px" viewBox="0 0 24 24" class="eva eva-trash-2" fill="#fff"><g data-name="Layer 2"><g data-name="trash-2"><rect width="24" height="24" opacity="0"/><path d="M21 6h-5V4.33A2.42 2.42 0 0 0 13.5 2h-3A2.42 2.42 0 0 0 8 4.33V6H3a1 1 0 0 0 0 2h1v11a3 3 0 0 0 3 3h10a3 3 0 0 0 3-3V8h1a1 1 0 0 0 0-2zM10 16a1 1 0 0 1-2 0v-4a1 1 0 0 1 2 0zm0-11.67c0-.16.21-.33.5-.33h3c.29 0 .5.17.5.33V6h-4zM16 16a1 1 0 0 1-2 0v-4a1 1 0 0 1 2 0z"/></g></g></svg>",
                }
              }
              style={
                {
                  "height": "20px",
                  "lineHeight": "20px",
                }
              }
            />
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
